### DISCUSSION TITLE
Biblatex-Chicago: How do I add a "lista" field for "inreference" or dictionary entries?

### DISCUSSION THREAD

**Discussion by ReaderGuy42** [retorquere/zotero-better-bibtex#2406] [Q&A] [answered]
I'm using LaTeX to format my citations, and I need to format my dictionary entries in a specific way. I have been told (here: https://tex.stackexchange.com/questions/674611/biblatex-chicago-help-with-correct-formatting-for-citing-dictionary-entry/674615#674615) that I need to use a field called `lista` to format the dictionary entries that way.

Is there any way for Zotero or Better Bibtex to add this field? I've tried the Extra field, but either it doesn't work or I'm not doing it right.

Any advice appreciated!

**Comment by retorquere** [comment, marked as answer]
You can add `tex.lista: something` to the `extra` field, and that will export. But is this a general biblatex feature, or specific to the chicago style?

**Reply by ReaderGuy42** [reply]
unfortunately that just outputs the citekey as the citation. I think this is an issue with how chicago style requires the formatting and how biblatex handles it? 
Any other ideas?
Thanks!!

**Reply by retorquere** [reply]
Can you send a debug log for the Zotero item, and post the desired biblatex output here?

**Reply by ReaderGuy42** [reply]
I'm sorry, I had missed that when I removed the title of the Zotero Item, it had (of course) changed the citekey. I didn't know it did that, though it makes sense. 
Therefore I'm not sure if a debug log would be useful since there's no problem as such? And I'm not sure what you mean by desired biblatex output exactly. This is the example of a working .bib entry that someone helped me with:

```
@inreference{slavish,
  title        = {Merriam-Webster},
  lista        = {slavish},
  url          = {https://www.merriam-webster.com/dictionary/slavish},
  urldate      = {2022-07-25},
}
```
Is this what you meant? Thanks!!

**Reply by retorquere** [reply]
You have a Zotero item in your library (the source item) you want to transform to this `@inreference` sample (the target). If this is indeed the target sample, you should have a source item that has a title `Merriam-Webster`. I need you to right-click that item and send a debug log. Please trust me when I say that I need it.

**Reply by ReaderGuy42** [reply]
Apologies I just found what you meant. Here's the code: 4F5SBWYI-refs-euc
Thanks so much!

**Reply by retorquere** [reply]
> I'm sorry, I had missed that when I removed the title of the Zotero Item, it had (of course) changed the citekey. I didn't know it did that, though it makes sense. 

If you pin the citekey, the citekey won't change anymore when you edit the item.

**Reply by ReaderGuy42** [reply]
Another cool function, thanks!

**Reply by retorquere** [reply]
You can have it pinned automatically after a delay: https://retorque.re/zotero-better-bibtex/installation/preferences/#automatically-pin-citation-key-after

**Reply by ReaderGuy42** [reply]
Unfortunately it's not letting me select that option to have it automatically pinned.

**Reply by retorquere** [reply]
`4F5SBWYI-refs-euc` exports to this by default:

```
@inreference{Merriam-Webster.comDictionarySlavish,
  title = {Slavish},
  author = {{Merriam-Webster.com Dictionary}},
  url = {https://www.merriam-webster.com/dictionary/slavish},
  urldate = {2022-07-25},
  abstract = {of or characteristic of someone held in forced servitude; especially : basely or abjectly servile; despicable, low; oppressive, tyrannical… See the full definition},
  langid = {english},
  keywords = {Secondary}
}
```

and you want this to be

```
@inreference{slavish,
  title        = {Merriam-Webster.com Dictionary},
  lista        = {slavish},
  url          = {https://www.merriam-webster.com/dictionary/slavish},
  urldate      = {2022-07-25},
}
```

You can do that either with a postscript:

```
if (Translator.BetterBibLaTeX && zotero.itemType == 'dictionaryEntry') {
  tex.add({ name: 'lista', value: zotero.title })
  tex.add({ name: 'title', value: zotero.creators.map(c => c.name || c.lastName).join(', ') })
  tex.remove('author')
}
```

to do it for all items, or by adding this to the `extra` field of the item:

```
tex.title: Merriam-Webster.com Dictionary
tex.author:
tex.lista: slavish
```

**Reply by retorquere** [reply]
> Unfortunately it's not letting me select that option to have it automatically pinned.

You can set it using the slider.

**Reply by ReaderGuy42** [reply]
> You can set it using the slider.

oh haha I thought it was a field that needed to be active or not

**Reply by ReaderGuy42** [reply]
> ```
> tex.title: Merriam-Webster.com Dictionary
> tex.author:
> tex.lista: slavish
> ```

So that postscript doesn't seem to do anything, (I pasted it into the postscript field under Preferencey > Export > postscript, is that correct?). But if I put the above into the extra field, it works!! So I'm happy. Thank you so much :)

**Reply by retorquere** [reply]
The UI is easily the weakest part of BBT. It's not my forte.

**Reply by ReaderGuy42** [reply]
That's fine, still awesome what you're doing!

**Reply by retorquere** [reply]
> So that postscript doesn't seem to do anything, (I pasted it into the postscript field under Preferencey > Export > postscript, is that correct?). But if I put the above into the extra field, it works!! So I'm happy. Thank you so much :)

Did you export as bibtex, or as biblatex? This particular postscript only works for biblatex (you mentioned biblatex-chicago, so I assumed you were using biblatex output).

**Reply by ReaderGuy42** [reply]
Under automatic export it says this:
![image](https://user-images.githubusercontent.com/71895400/218048005-c963405b-ae68-4a75-84e1-177815c2a50e.png)
so I must have selected BibLateX

**Reply by retorquere** [reply]
Then there must be an error somewhere, because this postscript should have worked. Can you enable debug logging from the help menu, re-export this automatic export, and send a debug log?

**Reply by ReaderGuy42** [reply]
D1604805200 this is the code that's given, but is that for this plugin or Zotero? Can you do anything with it?

**Reply by retorquere** [reply]
No, that's for Zotero. To get a BBT debug log, the menu entry is "Send Better-BibTeX debug log...", and the log ID will have `-euc` or `-apse` in it.

**Reply by ReaderGuy42** [reply]
OK, but I thought you wanted a log for the whole export? Or am I misunderstanding that?

**Reply by retorquere** [reply]
As long as the autoexport is triggered, and you send a BBT log after that, I'm good. Doesn't matter what items you have selected.

**Reply by ReaderGuy42** [reply]
OK, well then here's another code :)
UDAKQHTE-refs-euc
